docs/source/plugin/bot/channels.rst

```diff
@@ -0,0 +1,146 @@
+* its :attr:`~sopel.tools.target.Channel.topic`
+* its :attr:`~sopel.tools.target.Channel.users`
+* and its users' :attr:`~sopel.tools.target.Channel.privileges`
+
+.. note::
+
+    To check if a message comes from a channel, you have two options:
+
+    1. use the :func:`~sopel.plugin.require_chanmsg` decorator on your plugin
+       callable
+    2. use an ``if`` block in your function to check
+       :attr:`trigger.sender <sopel.trigger.Trigger.sender>`::
+
+           if not trigger.sender.is_nick():
+               # this trigger is from a channel
+
+       See :meth:`Identifier.is_nick() <sopel.tools.Identifier.is_nick>` for
+       more information.
+
+Getting users in a channel
+==========================
+
+To get a list of users in a **channel**, you can use its
+:attr:`~sopel.tools.target.Channel.users` attribute: this is a map of users you
+can iterate over to get all the users::
+
+    for nick, user in channel.users.items():
+        # do something with the nick and the user
+
+You can access one user in a channel with its nick::
+
+    user = channel.users['Nickname']
+
+With the ``trigger`` object, you can also access the user object directly::
+
+    user = channel.users[trigger.nick]
+
+Manage channels
+===============
+
+Like any IRC client, Sopel uses IRC messages to join or leave channels, to kick
+users or to set channel modes. It provides methods you can use to send those
+commands:
+
+* :meth:`bot.join() <sopel.irc.AbstractBot.join>`: to join a channel
+* :meth:`bot.part() <sopel.irc.AbstractBot.part>`: to leave a channel
+* :meth:`bot.kick() <sopel.irc.AbstractBot.kick>`: to kick a user from a
+  channel (this may requires some extra privileges)
+* :meth:`bot.write() <sopel.irc.AbstractBot.write>`: to send any message to IRC
+
+Join & Part
+-----------
+
+If you want the bot to join or leave a channel, the solution is
+straightforward::
+
+    bot.join('#channel_name')
+    bot.part('#channel_name')
+
+For example, you can recreate the ``.join #channel`` command like so::
+
+    @plugin.commands('join')
+    @plugin.require_admin()
+    def my_custom_join(bot, trigger):
+        channel, key = trigger.group(3), trigger.group(4)
+        bot.join(channel, password=key)
+
+In this example, the bot sends the following IRC message to the server:
+
+.. code-block:: irc
+
+    JOIN #channel
+
+When the server replies to that message, Sopel will automatically update its
+list of known channels and may send other request to know more about its users.
+
+However, it won't add the channel to its configuration: you will have to do
+that yourself by adding it to
+:attr:`bot.settings.core.channels <sopel.config.core_section.CoreSection.channels>`
+if you choose to do so.
+
+The ``part`` method works exactly like the ``join`` method: it sends a message
+to the IRC server and it is up to you to decide if you want to save or not
+that information in the settings—in that case by removing it from the list.
```

Comment:
Before I nitpick the wording here (I'm not fond of "to save or not that information"), there's a structural consideration to think about: Most plugins won't need to worry about joining/parting channels, and certainly won't have to deal with managing the bot's settings.

Could these notes (here and the above paragraph) about that be pulled into an aside instead? I'm thinking `.. note::` in this case, though one could also justify it being `.. important::`. Just a brief overview of these methods' existence and how they work is enough for the main text.